Index: mt7620-p4rev-113050/src/os/linux/rt_rbus_pci_drv.c
===================================================================
--- mt7620-p4rev-113050.orig/src/os/linux/rt_rbus_pci_drv.c
+++ mt7620-p4rev-113050/src/os/linux/rt_rbus_pci_drv.c
@@ -773,6 +773,13 @@ static void ac0_dma_done_tasklet(unsigne
 }
 
 
+#ifdef RALINK_ATE
+static void ate_ac0_dma_done_tasklet(unsigned long data)
+{
+	return;
+}
+#endif /* RALINK_ATE */
+
 
 #ifdef UAPSD_SUPPORT
 /*
